go/ql/lib/semmle/go/controlflow/ControlFlowGraph.qll

```diff
@@ -166,11 +166,21 @@ module ControlFlow {
       )
     }
 
+    /**
+     * Holds if this node writes `rhs` to `channel`.
+     */
+    predicate writesToChannel(DataFlow::Node channel, DataFlow::Node rhs) {
+      exists(SendStmt send |
+        send.getChannel() = channel.(DataFlow::ExprNode).asExpr() and
+        send.getValue() = rhs.(DataFlow::ExprNode).asExpr()
```

Comment:
Can you give the parameters type `DataFlow::ExprNode` and avoid the need to cast these?